### DISCUSSION TITLE
Is there a symbol server I can use for Roslyn assemblies deployed in VS?

### DISCUSSION THREAD

**Discussion by jnm2** [dotnet/roslyn#52030] [Q&A]
When debugging an analyzer (or even a VS extension in general), if I want to step into Microsoft.CodeAnalysis.dll etc, is there a symbol server that has the PDBs for any specific version that has shipped in VS? For example, I'm using 16.9.2 which says it comes with Roslyn version 3.9.0-6.21160.10+59eedc33d35754759994155ea2f4e1012a9951e3.

**Comment by jasonmalinowski** [comment]
@tmat, we upload these to a symbol server correct? Is it the standard Microsoft public one or some other one?

**Reply by jmarolf** [reply]
https://msdl.microsoft.com/download/symbols should have all roslyn symbols

**Reply by sharwell** [reply]
Yes, the public symbol server will have all Roslyn symbols, though there can be a publication delay after the product ships.

If the symbols are located but source stepping fails, it's possible msdl.microsoft.com only contains Windows PDB symbol files that are converted from Roslyn's reference portable PDBs with SourceLink information stripped out. @AArnott and I observed this problem trying to debug through some of his code in the past.

**Reply by jmarolf** [reply]
I did the following in Tools -> Options

1. Enabled symbol server support:
![image](https://user-images.githubusercontent.com/9797472/112040051-f2394600-8b01-11eb-889f-3cf9d9053663.png)
2. Checked the Microsoft symbol server source
![image](https://user-images.githubusercontent.com/9797472/112040078-f9605400-8b01-11eb-813d-ad1bb47ed473.png)
3. Disabled "Just My Code"
![image](https://user-images.githubusercontent.com/9797472/112040264-3b899580-8b02-11eb-9e4d-45edc4a757c3.png)


Attaching to the 16.9 devenv.exe process I get all the symbols to load in the modules window
![image](https://user-images.githubusercontent.com/9797472/112040813-e26e3180-8b02-11eb-96d0-0ee6974c7da4.png)
